### PR TITLE
Port "Fix chat" button

### PR DESCRIPTION
**Pull request by SirRichardFrancis**
## About The Pull Request

Press in case fancy chat didn't load.
Ported from downstream who ported it from TG.

## Why It's Good For The Game

QoL stuff.

## Changelog
:cl:
add: button for reloading fancy chat
/:cl:

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
Ok so on the drownstream it works like this -
You have to fail successfully
And tell it to swap to old chat, then fix chat again.

**Review by TheShown911 — COMMENTED** [review]
Maybe try making the notice messages sound more "professional". Who is this "me" that the player is talking to? The game client?

**Comment by SirRichardFrancis** [comment]
> Maybe try making the notice messages sound more "professional". Who is this "me" that the player is talking to? The game client?

TG coder, obviously.
image

**Review by hyperioo — APPROVED** [review]

File: code/game/verbs/ooc.dm [hunk 1]
```diff
@@ -174,3 +174,85 @@
...
+	set name = "Fix chat"
+	set category = "OOC"
+
+	if(!chatOutput || !istype(chatOutput))
+		var/action = alert(src, "Invalid Chat Output data found!\nRecreate data?", "", "Yes", "Cancel")
+		if(action != "Recreate Chat Output data")
+			return
+		chatOutput = new /datum/chatOutput(src)
+		chatOutput.start()
+		action = alert(src, "Goon chat is reloading.\nWait a bit and see if it's fixed.", "", "Fixed", "Nope")
```
Comment (hyperioo) [inline code/game/verbs/ooc.dm:188]:
```suggestion
		action = alert(src, "Goon chat is reloading.\nWait a bit and see if it\'s fixed.", "", "Fixed", "Nope")
```

File: code/game/verbs/ooc.dm [hunk 2]
```diff
@@ -174,3 +174,85 @@
...
+					winset(src, "browseroutput", "is-visible=false")
+				log_game("GOONCHAT: [key_name(src)] Failed to fix their goonchat window after recreating the chatOutput and forcing a load()")
+
+	else if(chatOutput.loaded)
+		var/action = alert(src, "Do you want to force a reload, wiping the chat log or just refresh the chat window?", "", "Force Reload", "Refresh", "Cancel")
+		switch (action)
+			if("Force Reload")
+				chatOutput.loaded = FALSE
+				chatOutput.start() //this is likely to fail since it asks , but we should try it anyways so we know.
+				action = alert(src, "Goon chat is reloading.\nWait a bit and see if it's fixed.", "", "Fixed", "Nope")
```
Comment (hyperioo) [inline code/game/verbs/ooc.dm:209]:
```suggestion
				action = alert(src, "Goon chat is reloading.\nWait a bit and see if it\'s fixed.", "", "Fixed", "Nope")
```

File: code/game/verbs/ooc.dm [hunk 3]
```diff
@@ -174,3 +174,85 @@
...
+					else
+						action = alert(src, "Try closing byond and reconnecting.\nCould also disable fancy chat and re-enable oldchat.", "", "Thanks anyways", "Switch to old chat")
+						if(action == "Switch to old chat")
+							winset(src, "output", "is-visible=true;is-disabled=false")
+							winset(src, "browseroutput", "is-visible=false")
+						log_game("GOONCHAT: [key_name(src)] Failed to fix their goonchat window forcing a start() and forcing a load()")
+
+			if("Refresh")
+				chatOutput.showChat()
+				action = alert(src, "Goon chat is refreshing.\nWait a bit and see if it's fixed.", "", "Fixed", "Nope")
```
Comment (hyperioo) [inline code/game/verbs/ooc.dm:226]:
```suggestion
				action = alert(src, "Goon chat is refreshing.\nWait a bit and see if it\'s fixed.", "", "Fixed", "Nope")
```

File: code/game/verbs/ooc.dm [hunk 4]
```diff
@@ -174,3 +174,85 @@
...
+						action = alert(src, "Try closing byond and reconnecting.\nCould also disable fancy chat and re-enable oldchat.", "", "Thanks anyways", "Switch to old chat")
+						if(action == "Switch to old chat")
+							winset(src, "output", "is-visible=true;is-disabled=false")
+							winset(src, "browseroutput", "is-visible=false")
+						log_game("GOONCHAT: [key_name(src)] Failed to fix their goonchat window forcing a show() and forcing a load()")
+		return
+
+	else
+		chatOutput.start()
+		var/action = alert(src, "Manually loading Chat.\nWait a bit and see if it's fixed.", "", "Fixed", "Nope")
```
Comment (hyperioo) [inline code/game/verbs/ooc.dm:245]:
```suggestion
		var/action = alert(src, "Manually loading Chat.\nWait a bit and see if it\'s fixed.", "", "Fixed", "Nope")
```

**Pull request merged**